examples/non-fungible-token/nft/src/lib.rs

```diff
@@ -0,0 +1,330 @@
+/*!
+Non-Fungible Token implementation with JSON serialization.
+NOTES:
+  - The maximum balance value is limited by U128 (2**128 - 1).
+  - JSON calls should pass U128 as a base-10 string. E.g. "100".
+  - The contract optimizes the inner trie structure by hashing account IDs. It will prevent some
+    abuse of deep tries. Shouldn't be an issue, once NEAR clients implement full hashing of keys.
+  - The contract tracks the change in storage before and after the call. If the storage increases,
+    the contract requires the caller of the contract to attach enough deposit to the function call
+    to cover the storage cost.
+    This is done to prevent a denial of service attack on the contract by taking all available storage.
+    If the storage decreases, the contract will issue a refund for the cost of the released storage.
+    The unused tokens from the attached deposit are also refunded, so it's safe to
+    attach more deposit than required.
+  - To prevent the deployed contract from being modified or deleted, it should not have any access
+    keys on its account.
+*/
+use near_contract_standards::non_fungible_token::metadata::{
+    NFTContractMetadata, NonFungibleTokenMetadataProvider, TokenMetadata, NFT_METADATA_SPEC,
+};
+use near_contract_standards::non_fungible_token::token::{Token, TokenId};
+use near_contract_standards::non_fungible_token::NonFungibleToken;
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::collections::LazyOption;
+use near_sdk::json_types::ValidAccountId;
+use near_sdk::{
+    env, near_bindgen, AccountId, BorshStorageKey, PanicOnDefault, Promise, PromiseOrValue,
+};
+use std::collections::HashMap;
```

Comment:
a weird side effect of the macro that requires importing this, should it not just use the reference itself?

Reply:
Are you highlighting the `use near_sdk::…` part? Curious what the suggestion here is. I see this is how we did it for the FT example, too.

Reply:
nah just the `HashMap` reference, since you don't directly use it in the code. Let me create a commit to show you what I mean

Reply:
Now I totally get it, great catch. I just moved the import your highlighted and another I just added (for `U128`) into the `macros.rs` file. So much better. 🤙🏼